gnoi/os/server_test.go

```diff
@@ -151,28 +179,196 @@ func TestTargetActivateAndVerify(t *testing.T) {
 }
 
 func TestTargetReceiveOS(t *testing.T) {
+	buf := make([]byte, 10000000)
+	rand.Read(buf)
+	oS := &mockos.OS{MockOS: mockosPb.MockOS{
+		Version: "1.0.2a",
+		Cookie:  "cookiestring",
+		Padding: buf,
+	}}
+	oS.Hash()
 	tests := []struct {
 		stream *mockTransferStream
 		err    error
 	}{
 		{
 			stream: &mockTransferStream{
-				responses: make(chan *pb.InstallResponse, 1),
+				response: make(chan *pb.InstallResponse, 1),
+				os:       oS,
 			},
 			err: nil,
 		},
 		{
 			stream: &mockTransferStream{
-				responses: make(chan *pb.InstallResponse, 1),
-				errorReq:  &pb.InstallRequest{Request: &pb.InstallRequest_TransferRequest{}}, // Unexpected request after transfer begins.
+				response: make(chan *pb.InstallResponse, 1),
+				errorReq: &pb.InstallRequest{Request: &pb.InstallRequest_TransferRequest{}}, // Unexpected request after transfer begins.
+				os:       oS,
 			},
 			err: errors.New("Unknown request type"),
 		},
 	}
 	for _, test := range tests {
+		test.stream.response <- &pb.InstallResponse{Response: &pb.InstallResponse_TransferReady{}}
 		_, err := ReceiveOS(test.stream)
 		if diff := pretty.Compare(test.err, err); diff != "" {
 			t.Errorf("ReceiveOS(stream): (-want +got):\n%s", diff)
 		}
 	}
 }
+
+func TestTargetInstall(t *testing.T) {
+	buf := make([]byte, 10000000)
+	rand.Read(buf)
+	oS := &mockos.OS{MockOS: mockosPb.MockOS{
+		Version: "1.0.2a",
+		Cookie:  "cookiestring",
+		Padding: buf,
+	}}
+	oS.Hash()
+	incompatibleOS := &mockos.OS{MockOS: mockosPb.MockOS{
+		Version:      "1.0.2b",
+		Cookie:       "cookiestring",
+		Padding:      buf,
+		Incompatible: true,
+	}}
+	incompatibleOS.Hash()
+	tests := []struct {
```

Comment:
make sure you have a string field called name or similar(this has to do with my next comment)about t.Run)